spring-cloud-consul-config/src/main/java/org/springframework/cloud/consul/config/watch/ConsulConfigWatchConfiguration.java

```diff
@@ -0,0 +1,36 @@
+package org.springframework.cloud.consul.config.watch;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.endpoint.RefreshEndpoint;
+import org.springframework.cloud.consul.ConsulAutoConfiguration;
+import org.springframework.cloud.consul.config.ConsulConfigProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+@Configuration
+@Import(ConsulAutoConfiguration.class)
+@EnableConfigurationProperties
+@EnableScheduling
+@ConditionalOnProperty(name = "spring.cloud.consul.config.watch", matchIfMissing = false)
+public class ConsulConfigWatchConfiguration {
```

Comment:
I think I would put this all in `ConsulConfigBootstrapConfiguration` rather than auto-configuration.